examples/session-flash/app/routes/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { useEffect, useRef } from "react";
+import { FlashMessage, getSession, storage } from "~/utils/session.server";
+import { getSessionFlash } from "~/utils/session.server";
+
+interface LoaderData {
+  message?: FlashMessage;
+}
+
+interface ActionData {
+  formError: string;
+}
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const flash = await getSessionFlash(request);
+  if (flash && flash.message) {
+    return json({ message: flash.message }, { headers: flash.headers });
+  }
+
+  return null;
+};
+
+export const action: ActionFunction = async ({ request }) => {
+  const form = await request.formData();
+  const text = form.get("messageText");
+  const color = form.get("messageColor");
+
+  if (!text || !color) {
+    return json({ formError: `Invalid form submission` }, { status: 400 });
+  }
+
+  const session = await getSession(request);
+  session.flash("messageText", text);
+  session.flash("messageColor", color);
+
+  return redirect(".", {
+    headers: { "Set-Cookie": await storage.commitSession(session) },
+    status: 200,
+  });
+};
+
+export default function Index() {
+  const loaderData = useLoaderData<LoaderData>();
+  const actionData = useActionData<ActionData>();
+  return (
+    <>
+      {loaderData?.message ? (
+        <FlashMessage message={loaderData.message} />
+      ) : null}
+      <Form method="post">
+        <label htmlFor="messageText">Enter your message: </label>
+        <input
+          id="messageText"
+          name="messageText"
+          type="text"
+          defaultValue="This is a flash message ⚡"
+          required
+        />
+        <br />
+
+        <fieldset style={{ width: "fit-content" }}>
+          <legend>Pick a color</legend>
+          <input
+            id="info"
+            name="messageColor"
+            type="radio"
+            value="#11bddf"
+            defaultChecked
+          />
+          <label htmlFor="info">info</label>
+
+          <input
+            id="success"
+            name="messageColor"
+            type="radio"
+            value="#1bcd18"
+          />
+          <label htmlFor="success">success</label>
+
+          <input id="error" name="messageColor" type="radio" value="#e81717" />
+          <label htmlFor="error">error</label>
+        </fieldset>
+
+        <br />
+        <button type="submit">Set Flash Message</button>
+        {actionData?.formError ? (
+          <pre style={{ color: "red" }}>{actionData.formError}</pre>
+        ) : null}
+      </Form>
+    </>
+  );
+}
+
+function FlashMessage({ message }: { message: FlashMessage }) {
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    if (message && !dialogRef.current?.open) dialogRef.current?.show();
+  }, [message]);
+
+  return message ? (
+    <dialog ref={dialogRef} id="dialog" style={{ borderColor: message.color }}>
+      <p>{message.text}</p>
+      <form method="dialog">
+        <button>Close</button>
+      </form>
```

Comment:
looks like there is https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-method 🤯

Reply:
Holy smokes 🤯

Reply:
Thanks for accepting my modest contribution 🙏 I went a bit overboard with the #useThePlatform thing 😋

I wanted to use the opportunity of my example to make other people learn surprising things on the way.

I wasn't sure about <dialog>, as it isn't supported by Typescript yet (I had to import my own interface). But it's definitely nice to know a native way to create dialogs and modals without relying on Javascript. 

In my case, I was more mind-blowned by [`color-scheme: dark light`](https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme) which use the browser for applying native dark theme (if preferred) without any additionnal CSS 🤯

Should I add a section to explain my Easter eggs?

Reply:
Actually I almost asked you to remove that bit because we normally want to keep things as simple and focused as possible. I don't think we need to explain those things. Thanks though!

Reply:
Thank you then. I'll try to keep it more concise next time! I just didn't want to reproduce the exact same example as the one already in your docs 😋

I'll see where my current project takes me, I would love to contribute again 🔥